Topics/Development_Process/Shell/intro_to_shell.md

```diff
@@ -0,0 +1,159 @@
+# Detailed Introduction to Shell Commands
+
+Shell commands are the primary interface for users to interact with an operating system's services in a Unix or Unix-like environment. These commands are entered into a command-line interface (CLI), such as Bash in Linux or Command Prompt in Windows.
+
+## What is a Shell?
+
+The shell is a program that takes commands from the keyboard and gives them to the operating system to perform. It's a powerful tool that allows users to perform complex tasks with simple commands.
+
+## Types of Shells
+
+- **Bash (Bourne Again SHell):** The most common shell in Linux distributions.
+- **Tcsh/Csh (TENEX/TOPS C Shell):** Known for its scripting capabilities.
+- **Ksh (Korn Shell):** A shell combining features of Bash and Csh.
+- **Zsh (Z Shell):** Offers improvements over Bash, including themes and plugins.
+
+Understanding different shells helps you choose the right one for your needs and understand the syntax differences in scripting.
+
+## Why Learn Shell Commands?
+
+- **Automation:** Automate repetitive tasks with scripts.
+- **Efficiency:** Perform tasks faster than using a graphical interface.
+- **Control:** Gain detailed control over system functions.
+
+Learning shell commands is an essential skill for software engineers, as it boosts productivity and efficiency in managing systems and developing software.
+
+
+# Navigating the File System
+
+Understanding how to navigate the file system using the command line is a fundamental skill. Here's how to get started:
+
+## Understanding File System Structure
+
+- **Root Directory:** Denoted by `/`, it's the top-level directory.
+- **Home Directory:** Denoted by `~`, it's the user's personal directory.
+- **Relative Path:** Specifies a location relative to the current directory.
+- **Absolute Path:** Specifies a location from the root directory.
+
+## Basic Navigation Commands
+
+- `pwd`: Print Working Directory. Shows your current directory.
```

Comment:
For these commands, do you want to include an image perhaps? 
So that it's easier for someone who doesn't know shell to follow along what you're saying

Reply:
Thanks for the suggestion! Added a image to demonstrate the usage of these basic commands.